daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala

```diff
@@ -472,15 +470,17 @@ private[archive] class DecodeV1(minor: LV.Minor) {
         "EnumConstructors.constructors",
       )
 
-    private[this] def decodeDefValue(lfValue: PLF.DefValue): DValue = {
+    private[lf] def decodeDefValue(lfValue: PLF.DefValue): DValue = {
+      if (!lfValue.getNoPartyLiterals) {
+        throw Error.Parsing("DefValue must have no_party_literals set to true")
```

Comment:
@cocreature @remyhaemmerle-da : throwing this parsing error changes a handful of canton test based on the "illformed.dar" to no longer generate a "standard"/"self-service" package validation error anymore according to canton-ci. e.g.:

https://app.circleci.com/pipelines/github/DACH-NY/canton/29790/workflows/9b403b83-fe5b-4078-9a56-0cc86bc6aef4/jobs/595170

Shouldn't this be a package validation error, too?

cc - @andreaslochbihler-da

Reply:
@remyhaemmerle-da correct me if I’m wrong, my understanding is that we throw different errors for decoding than we throw for validation (e.g. typechecking). So this changes it to throw a decoding error instead of a package validation error. If we don’t have self-service errors for decoding failures we should fix that.

Reply:
In particular even without this change you can have decoding failures so we need to handle them regardless.

Reply:
Makes sense in general, but I think with these tests in particularly we were looking for a dar (provided by Remy?) that triggered a package validation problem, so we could verify that we properly handle validation errors in canton.

Reply:
Yes, we wanted to get a type checking error with `illformed.dar`. Probably it suffices to regenerate the dar such that the `getNoPartyLiterals` is set. @remyhaemmerle-da Can you regenerate such a dar?

Reply:
sent the DAR on slack